<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>HexEditorWindow</class>
 <widget class="QMainWindow" name="HexEditorWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>728</width>
    <height>611</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Hex Editor</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="spacing">
     <number>4</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="hexTab">
       <attribute name="title">
        <string>Hex Editor</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QSplitter" name="mainSplitter">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <widget class="HexEditor" name="leftFilePanel">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
          <widget class="HexEditor" name="rightFilePanel">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
          <widget class="QWidget" name="verticalLayoutWidget">
           <layout class="QVBoxLayout" name="verticalLayout">
            <property name="sizeConstraint">
             <enum>QLayout::SetDefaultConstraint</enum>
            </property>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <property name="spacing">
               <number>4</number>
              </property>
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QPushButton" name="newStructureButton">
                <property name="maximumSize">
                 <size>
                  <width>50</width>
                  <height>16777215</height>
                 </size>
                </property>
                <property name="text">
                 <string>New</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="openStructureButton">
                <property name="maximumSize">
                 <size>
                  <width>50</width>
                  <height>16777215</height>
                 </size>
                </property>
                <property name="text">
                 <string>Open</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="saveStructureButton">
                <property name="maximumSize">
                 <size>
                  <width>50</width>
                  <height>16777215</height>
                 </size>
                </property>
                <property name="text">
                 <string>Save</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="saveAsStructreButton">
                <property name="maximumSize">
                 <size>
                  <width>50</width>
                  <height>16777215</height>
                 </size>
                </property>
                <property name="text">
                 <string>Save As</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QLabel" name="structureFilename">
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QTextEdit" name="structureEdit">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="maximumSize">
               <size>
                <width>16777215</width>
                <height>16777215</height>
               </size>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_3">
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QPushButton" name="applyStructureButton">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Apply Structure</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="formatButton">
                <property name="text">
                 <string>Format Structure</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="structureTab">
       <attribute name="title">
        <string>Structure Editor</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="StructureEditor" name="structureEditor" native="true"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>728</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QMenu" name="menuOpen_Recent">
     <property name="title">
      <string>Open Recent</string>
     </property>
     <addaction name="actionNo_recent_items"/>
    </widget>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionSave_As"/>
    <addaction name="menuOpen_Recent"/>
    <addaction name="actionSave2"/>
    <addaction name="actionSave_As2"/>
    <addaction name="separator"/>
    <addaction name="actionPaste_File"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionCut"/>
    <addaction name="actionCopy"/>
    <addaction name="actionPaste"/>
    <addaction name="actionFind"/>
    <addaction name="actionLeftToRight"/>
    <addaction name="actionCopy_Right_To_Left"/>
   </widget>
   <widget class="QMenu" name="menuSettings">
    <property name="title">
     <string>Settings</string>
    </property>
    <addaction name="actionSettings"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Tools</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionHelp"/>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuStructure">
    <property name="title">
     <string>Structure</string>
    </property>
    <widget class="QMenu" name="menuOpen_Recent_2">
     <property name="title">
      <string>Open Recent</string>
     </property>
     <addaction name="actionNo_recent_items_2"/>
    </widget>
    <addaction name="actionNew_2"/>
    <addaction name="actionOpen_2"/>
    <addaction name="actionSave_2"/>
    <addaction name="actionSave_As_2"/>
    <addaction name="menuOpen_Recent_2"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuStructure"/>
   <addaction name="menuSettings"/>
   <addaction name="menuTools"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionNew"/>
   <addaction name="actionOpen"/>
   <addaction name="actionSave"/>
   <addaction name="actionSave_As"/>
   <addaction name="actionSave2"/>
   <addaction name="actionSave_As2"/>
   <addaction name="separator"/>
   <addaction name="actionCut"/>
   <addaction name="actionCopy"/>
   <addaction name="actionPaste"/>
   <addaction name="actionFind"/>
   <addaction name="separator"/>
   <addaction name="actionLeftToRight"/>
   <addaction name="actionCopy_Right_To_Left"/>
   <addaction name="separator"/>
   <addaction name="actionSettings"/>
   <addaction name="separator"/>
   <addaction name="actionExit"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionNew">
   <property name="icon">
    <iconset resource="structurehexeditor.qrc">
     <normaloff>:/main/icons/icons/New.png</normaloff>:/main/icons/icons/New.png</iconset>
   </property>
   <property name="text">
    <string>New</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="structurehexeditor.qrc">
     <normaloff>:/main/icons/icons/Open.png</normaloff>:/main/icons/icons/Open.png</iconset>
   </property>
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="structurehexeditor.qrc">
     <normaloff>:/main/icons/icons/Save.png</normaloff>:/main/icons/icons/Save.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionSave_As">
   <property name="icon">
    <iconset resource="structurehexeditor.qrc">
     <normaloff>:/main/icons/icons/SaveAs.png</normaloff>:/main/icons/icons/SaveAs.png</iconset>
   </property>
   <property name="text">
    <string>Save As</string>
   </property>
  </action>
  <action name="actionSave2">
   <property name="icon">
    <iconset resource="structurehexeditor.qrc">
     <normaloff>:/main/icons/icons/Save_2.png</normaloff>:/main/icons/icons/Save_2.png</iconset>
   </property>
   <property name="text">
    <string>Save2</string>
   </property>
  </action>
  <action name="actionSave_As2">
   <property name="icon">
    <iconset resource="structurehexeditor.qrc">
     <normaloff>:/main/icons/icons/SaveAs_2.png</normaloff>:/main/icons/icons/SaveAs_2.png</iconset>
   </property>
   <property name="text">
    <string>Save As2</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="icon">
    <iconset resource="structurehexeditor.qrc">
     <normaloff>:/main/icons/icons/Exit.png</normaloff>:/main/icons/icons/Exit.png</iconset>
   </property>
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionCopy">
   <property name="icon">
    <iconset resource="structurehexeditor.qrc">
     <normaloff>:/main/icons/icons/Copy.png</normaloff>:/main/icons/icons/Copy.png</iconset>
   </property>
   <property name="text">
    <string>Copy</string>
   </property>
  </action>
  <action name="actionPaste">
   <property name="icon">
    <iconset resource="structurehexeditor.qrc">
     <normaloff>:/main/icons/icons/Paste.png</normaloff>:/main/icons/icons/Paste.png</iconset>
   </property>
   <property name="text">
    <string>Paste</string>
   </property>
  </action>
  <action name="actionCut">
   <property name="icon">
    <iconset resource="structurehexeditor.qrc">
     <normaloff>:/main/icons/icons/Cut.png</normaloff>:/main/icons/icons/Cut.png</iconset>
   </property>
   <property name="text">
    <string>Cut</string>
   </property>
  </action>
  <action name="actionFind">
   <property name="icon">
    <iconset resource="structurehexeditor.qrc">
     <normaloff>:/main/icons/icons/Find.png</normaloff>:/main/icons/icons/Find.png</iconset>
   </property>
   <property name="text">
    <string>Find</string>
   </property>
  </action>
  <action name="actionLeftToRight">
   <property name="icon">
    <iconset resource="structurehexeditor.qrc">
     <normaloff>:/main/icons/icons/LeftToRight.png</normaloff>:/main/icons/icons/LeftToRight.png</iconset>
   </property>
   <property name="text">
    <string>Copy Left To Right</string>
   </property>
  </action>
  <action name="actionCopy_Right_To_Left">
   <property name="icon">
    <iconset resource="structurehexeditor.qrc">
     <normaloff>:/main/icons/icons/RightToLeft.png</normaloff>:/main/icons/icons/RightToLeft.png</iconset>
   </property>
   <property name="text">
    <string>Copy Right To Left</string>
   </property>
  </action>
  <action name="actionSettings">
   <property name="icon">
    <iconset resource="structurehexeditor.qrc">
     <normaloff>:/main/icons/icons/Settings.png</normaloff>:/main/icons/icons/Settings.png</iconset>
   </property>
   <property name="text">
    <string>Settings</string>
   </property>
  </action>
  <action name="actionHelp">
   <property name="icon">
    <iconset resource="structurehexeditor.qrc">
     <normaloff>:/main/icons/icons/Help.png</normaloff>:/main/icons/icons/Help.png</iconset>
   </property>
   <property name="text">
    <string>Help</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="icon">
    <iconset resource="structurehexeditor.qrc">
     <normaloff>:/main/icons/icons/About.png</normaloff>:/main/icons/icons/About.png</iconset>
   </property>
   <property name="text">
    <string>About ...</string>
   </property>
  </action>
  <action name="actionNo_recent_items">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>No recent items</string>
   </property>
  </action>
  <action name="actionPaste_File">
   <property name="text">
    <string>Paste File</string>
   </property>
  </action>
  <action name="actionNew_2">
   <property name="text">
    <string>New</string>
   </property>
  </action>
  <action name="actionOpen_2">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionSave_2">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionSave_As_2">
   <property name="text">
    <string>Save As</string>
   </property>
  </action>
  <action name="actionNo_recent_items_2">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>No recent items</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>HexEditor</class>
   <extends>QFrame</extends>
   <header>hexeditor.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>StructureEditor</class>
   <extends>QWidget</extends>
   <header>structureeditor.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="structurehexeditor.qrc"/>
 </resources>
 <connections/>
</ui>
